sdk/@launchdarkly/observability-react-native/src/instrumentation/errorTypes.ts

```diff
@@ -0,0 +1,82 @@
+import { Attributes } from '@opentelemetry/api'
+
+export type ErrorType =
+	| 'unhandled_exception'
+	| 'unhandled_rejection'
+	| 'console_error'
+	| 'react_error'
+export type ErrorSource = 'javascript' | 'native' | 'react'
+
+export interface ErrorContext {
+	type: ErrorType
+	source: ErrorSource
+	fatal: boolean
+	componentStack?: string
+	appState?: 'active' | 'background' | 'inactive'
+	timestamp: number
+	sessionId?: string
+}
+
+export interface FormattedError {
+	message: string
+	name: string
+	stack?: string
+	context: ErrorContext
+	attributes: Attributes
+}
+
+// Error deduplication
+export interface ErrorFingerprint {
+	message: string
+	name: string
+	stackHash: string
+}
+
+export class ErrorDeduplicator {
+	private recentErrors: Map<string, number> = new Map()
+	private readonly MAX_CACHE_SIZE = 100
+	private readonly DEDUP_WINDOW_MS = 5000 // 5 seconds
+
+	shouldReport(error: Error): boolean {
+		const fingerprint = this.getFingerprint(error)
+		const key = JSON.stringify(fingerprint)
```

Comment:
The JSON here can be non-deterministically sorted, so this will may not always work as expected.

Reply:
I have a canonical parser in js-core that could be used.